analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Sonar/SonarExplodedGraphTest.cs

```diff
@@ -901,6 +900,33 @@ public void ExplodedGraph_IsPattern_WithPositionalPattern()
             context.WalkWithInstructions(4);
         }
 
+        [DataTestMethod]
+        [DataRow("int")]
+        [DataRow("(string s)")]
+        [DataRow("1 or 2")]
+        [DataRow("1 and 2")]
+        [DataRow("not 1")]
+        [DataRow("> 5")]
+        [DataRow("[] empty")]
+        public void ExplodedGraph_SwitchStatement_UnsupportedPatternKinds(string pattern)
+        {
+            var testInput = $$"""
+                switch (new object())
+                {
+                    case {{pattern}}: break;
+                }
+                """;
+            var context = new ExplodedGraphContext(testInput);
+            var actualInstructions = new List<string>();
+            context.ExplodedGraph.InstructionProcessed += (sender, args) =>
+                actualInstructions.Add(args.Instruction.ToString());
```

Comment:
As we throw, we don't really care about the visited instructions for purpose of asserting them.
```suggestion
```